### PR TITLE
Diana resize

### PR DESCRIPTION
**Pull request by madisonmay**

### CODE REVIEW

File: tests/test_remote.py [hunk 1]
```diff
@@ -550,6 +550,11 @@ def test_invalid_int_numpy_arrays(self):
         test_image = np.random.randint(255, 300, size=(48,48, 5))
         self.assertRaises(IndicoError, image_features, test_image)
 
+    def test_numpy_resize_content_filtering(self):
+        test_image = np.random.randint(0, 255, size=(480,248, 3))
+        response = content_filtering(test_image)
+        self.assertTrue(isinstance(response, float))
+
```
Comment (madisonmay) [inline tests/test_remote.py]: Would also be nice to add a quick unittest for the `image_preprocess` function with the `min_axis` argument.  Doesn't need to be complicated, but a sanity check is super nice.

File: indicoio/utils/image.py [hunk 2]
```diff
@@ -50,6 +50,17 @@ def image_preprocess(image, size=(48,48), batch=False):
     # image resizing
     if size:
```
Comment (madisonmay) [inline indicoio/utils/image.py]: Would be super nice to pull out all this code related to resizing into another function.

File: indicoio/utils/image.py [hunk 3]
```diff
@@ -50,6 +50,17 @@ def image_preprocess(image, size=(48,48), batch=False):
...
         out_image = out_image.resize(size)
+    if min_axis:
+        min_idx, other_idx = (0,1) if out_image.size[0] < out_image.size[1] else (1,0)
+        size_arr = [0,0]
+        size_arr[min_idx] = min_axis
+        size_arr[other_idx] = int((min_axis/float(out_image.size[min_idx]))*out_image.size[other_idx])
+        out_image = out_image.resize(tuple(size_arr))
+        if size_arr[0]/float(size_arr[1]) > 10 or size_arr[0]/float(size_arr[1]) < 0.1:
```
Comment (madisonmay) [inline indicoio/utils/image.py]: Why check both when you can use `size_arr[min_idx]/float(size_array[other_idx])` (or vice versa) for the check?

File: indicoio/utils/image.py [hunk 4]
```diff
@@ -50,6 +50,17 @@ def image_preprocess(image, size=(48,48), batch=False):
     # image resizing
     if size:
         out_image = out_image.resize(size)
+    if min_axis:
+        min_idx, other_idx = (0,1) if out_image.size[0] < out_image.size[1] else (1,0)
+        size_arr = [0,0]
+        size_arr[min_idx] = min_axis
+        size_arr[other_idx] = int((min_axis/float(out_image.size[min_idx]))*out_image.size[other_idx])
```
Comment (madisonmay) [inline indicoio/utils/image.py]: Nitpicking at this point, but might be more readable if broken up into:

```python
scale = min_axis/float(out_image.size[min_idx])
size_arr[other_idx] = int(scale * out_image.size[other_idx])
```

**Comment by madisonmay** [comment]
Few minor edits but mostly looks good to me @dianavermilya

**Comment by madisonmay** [comment]
TODO: 
- [ ] Address edits
- [ ] Final code review
- [ ] Rebase 
- [ ] Merge

**Comment by dianavermilya** [comment]
Oh, I didn't realise you were going to review this.  (I didn't finish testing/do cleanup.)  Well, thanks for the suggestions.  Will do!

**Comment by dianavermilya** [comment]
@madisonmay ok this is ready for review whenever.

File: indicoio/images/filtering.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 from indicoio.utils.image import image_preprocess
 import indicoio.config as config
 
-def content_filtering(image, cloud=None, batch=False, api_key=None, **kwargs):
+def content_filtering(image, min_axis=128, cloud=None, batch=False, api_key=None, **kwargs):
```
Comment (madisonmay) [inline indicoio/images/filtering.py]: This shouldn't be exposed to the user -- it's an implementation detail that we use internally.  So we can use this value in `image_preprocess` but don't need it in the `content_filtering` parameters.

**Comment by madisonmay** [comment]
One final change and I think this is ready to merge in! @dianavermilya

**Comment by madisonmay** [comment]
Awesome stuff, rebase time :)

**Comment by madisonmay** [comment]
I think there was a regression after the rebase?  Looks like we kept some of the older versions...

**Comment by dianavermilya** [comment]
@madisonmay should be all good to merge

**Comment by madisonmay** [comment]
Looks great! Merging...

File: indicoio/utils/image.py [hunk 6]
```diff
@@ -54,6 +52,25 @@ def image_preprocess(image, size=(48,48), batch=False):
     return base64.b64encode(output_s)
 
 
+def resize_image(image, size, min_axis):
+    if size:
```
Comment (sihrc) [inline indicoio/utils/image.py]: @madisonmay

Comment (sihrc) [inline indicoio/utils/image.py]: We should be doing one or the other right? so maybe check for min_axis first and then else if for size or something?

**Pull request merged**